pkg/util/restmapper/restmapper.go

```diff
@@ -65,16 +67,49 @@ func (g *cachedRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string)
 	}
 
 	// consult underlying mapper and then update cache
-	restMapping, err := g.restMapper.RESTMapping(gk, versions...)
+	restMapping, err := g.getMapper().RESTMapping(gk, versions...)
+	if meta.IsNoMatchError(err) {
+		// hit here means a resource might be missing from the current rest mapper,
+		// probably because a new resource(CRD) has been added, we have to reload
+		// resource and rebuild the rest mapper.
+
+		var groupResources []*restmapper.APIGroupResources
+		groupResources, err = restmapper.GetAPIGroupResources(g.discoveryClient)
+		if err != nil {
+			return nil, err
+		}
+
+		newMapper := restmapper.NewDiscoveryRESTMapper(groupResources)
+		restMapping, err = newMapper.RESTMapping(gk, versions...)
+		if err == nil {
+			// hit here means after reloading, the new rest mapper can recognize
+			// the resource, we have to replace the mapper and clear cache.
+			g.mu.Lock()
+			g.restMapper = newMapper
+			g.mu.Unlock()
+			g.gvkToGVR.Range(func(key, value any) bool {
```

Comment:
Just a question.Why not directly use `g.gvkToGVR.Delete(key)`?

Reply:
The logic here is going to `clear` the whole cache, not just remove a specific key. 
For example, after removing a CRD, the key may remain in the cache(gvkToGVR), so the idea is to clear the cache and then rebuild the cache with the latest loaded resources.